x-pack/platform/plugins/shared/fleet/server/services/setup/update_deprecated_component_templates.ts

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import pMap from 'p-map';
+import type { ElasticsearchClient } from '@kbn/core/server';
+
+import { appContextService } from '..';
+
+export async function updateDeprecatedComponentTemplates(esClient: ElasticsearchClient) {
+  const componentTemplates = await esClient.cluster.getComponentTemplate({
```

Comment:
Should we limit this to `metrics-apm.*` as I think we want to avoid updating user component template?

Reply:
we could, though I'm not sure if this can happen for other packages too

Reply:
Maybe we could filter for component template `managed_by:fleet` in the meta?

Reply:
good idea, added it as a filter